modules/httplib/url.go

```diff
@@ -26,7 +32,60 @@ func IsRelativeURL(s string) bool {
 	return err == nil && urlIsRelative(s, u)
 }
 
-func IsCurrentGiteaSiteURL(s string) bool {
+func guessRequestScheme(req *http.Request) string {
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto
```

Comment:
The link mentions the [`Forwarded`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded) header. Maybe add this too?

Reply:
`Forwarded` header has a very complex syntax, and it isn't designed for this case.

And I have covered the headers in our document: https://docs.gitea.com/administration/reverse-proxies#nginx

```
        proxy_set_header Host $host;
        proxy_set_header X-Forwarded-Proto $scheme;
```